### ISSUE TITLE
Is it necessary to load the predictor to use exported onnx model?

### ISSUE THREAD

**Issue by LiShaoyu5** [autogluon/autogluon#4030]
I am working on deploying a text model trained by MultiModalPredictor and exported it to onnx by:
`ag_model.export_onnx("./export/data.csv", onnx_model_path)`

Then I referred to the [example](https://github.com/autogluon/autogluon/blob/master/examples/automm/production/onnx_text.py), which processes input like this:
```
def eval_cosine(predictor, df, onnx_session):
    labels = df["score"].to_numpy()
    valid_input = [
        "hf_text_text_token_ids",
        "hf_text_text_valid_length",
        "hf_text_text_segment_ids",  # Remove for mpnet
    ]

    QEmb = onnx_session.run(None, predictor._learner.get_processed_batch_for_deployment(data=df[["sentence1"]]))[0]
    AEmb = onnx_session.run(None, predictor._learner.get_processed_batch_for_deployment(data=df[["sentence2"]]))[0]

    cosine_scores = 1 - (paired_cosine_distances(QEmb, AEmb))
    eval_pearson_cosine, _ = pearsonr(labels, cosine_scores)
    eval_spearman_cosine, _ = spearmanr(labels, cosine_scores)

    print(eval_pearson_cosine)
    print(eval_spearman_cosine)
```
It uses `predictor._learner.get_processed_batch_for_deployment` to process the input text. 

I want the deployment environment to be as small as possible, Can I just use the transfomers.AutoTokenizer to process the input? What is the relationship between hf tokenizer's output: ['input_ids', 'attention_mask', 'token_type_ids'] and autogluon's: ['hf_text_text_token_ids', 'hf_text_text_segment_ids', 'hf_text_text_valid_length']

Update:

Here is how to process the input to get the onnx model running:
```
# example
inputs = ag_model._learner.get_processed_batch_for_deployment(data=pd.DataFrame({"text": ["test_data"]})[["text"]])

# same with tokenizer and numpy
inputs = tokenizer(
        "test_data",
        padding="do_not_pad",
        max_length=2048,  # for my model
        truncation=True,
        return_tensors="np"
    )
inputs = {
        "hf_text_text_token_ids": inputs["input_ids"],
        "hf_text_text_valid_length": np.array([inputs["attention_mask"].shape[1]]),
        "hf_text_text_segment_ids": inputs["token_type_ids"],
    }

output = onnx_model.run(None, inputs)[0]  # works for both
```

**Comment by LiShaoyu5** [comment]
Or, how can I make a minimal installation just for inference(as in the example, only to load the functions to process the input text)? Installing autogluon.muitimodal requires too many packages, especially torch, which will make the environment too big. My original expectation for the environment just contains onnx things.

**Comment by LiShaoyu5** [comment]
After using ag_model._learner.get_processed_batch_for_deployment to get the script running, I got even more confused. Here is the inference code:
`output = model.run(None, ag_model._learner.get_processed_batch_for_deployment(data=pd.DataFrame({"text": ["test_text"]})[["text"]]))[0]`

The output is an array of shape (1, 512). But the model I trained has only 134 labels, which is correct in `ag_model.class_labels`.

**Comment by LiShaoyu5** [comment]
> @LiShaoyu5 I see. So we still need to have this dependency "predictor._learner.get_processed_batch_for_deployment"? If would be perfect if we can just remove this function even.

no need, the second way in mt update is enough for inference so the minimal requirement should be transformers,  numpy and onnxruntime.
